File: java/meal-planner-backend/src/main/java/com/meal_planner/meal_planner_backend/service/MealService.java
package com.meal_planner.meal_planner_backend.service;

import com.meal_planner.meal_planner_backend.model.Meal;
import com.meal_planner.meal_planner_backend.repository.MealRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Map;

@Service
public class MealService {

    @Autowired
    private MealRepository mealRepository;

    public Map<String, String> getAllMeals() {
        return mealRepository.getMealPlan().getMeals();
    }

    public void updateMeal(String day, String meal) {
        mealRepository.updateMeal(day, meal);
    }

    public void deleteMeal(String day) {
        mealRepository.updateMeal(day, "");
    }
}

File: csharp/MealPlanner.Infrastructure/Services/MealService.cs
using MealPlanner.Core.Interfaces;

namespace MealPlanner.Infrastructure.Services;

public class MealService : IMealService
{
    private readonly IMealRepository _mealRepository;

    public MealService(IMealRepository mealRepository)
    {
        _mealRepository = mealRepository;
    }

    public Dictionary<string, string> GetAllMeals()
    {
        return _mealRepository.GetMealPlan().GetMeals();
    }

    public void UpdateMeal(string day, string meal)
    {
        _mealRepository.UpdateMeal(day, meal);
    }

    public void DeleteMeal(string day)
    {
        _mealRepository.UpdateMeal(day, string.Empty);
    }
}